### ISSUE TITLE
(0 , _reactNative.TabNavigator) is not a function

### ISSUE THREAD

**Issue by arnoldbird** [expo/expo#1596]
### Environment

Environment:
  OS: macOS High Sierra 10.13.3
  Node: 9.5.0
  Yarn: 1.3.2
  npm: 5.6.0
  Watchman: 4.9.0
  Xcode: Xcode 9.2 Build version 9C40b
  Android Studio: Not Found

Packages: (wanted => installed)
  expo: ^25.0.0 => 25.0.0
  react: 16.2.0 => 16.2.0
  react-native: https://github.com/expo/react-native/archive/sdk-25.0.0.tar.gz => 0.52.0

Diagnostics report:
  https://exp-xde-diagnostics.s3.amazonaws.com/arnoldbird-accb5143-f337-4fde-809e-2ed974c91ab0.tar.gz

### Steps to Reproduce

I am trying to move src code from a project into a new expo project.  When I run the simulator, I see an error...

> (0 , _reactNative.TabNavigator) is not a function

(Note that this is the same error in [this issue](https://github.com/react-navigation/react-navigation/issues/812) from last year.)

To reproduce:
1) Create a new app as described in the CRNA readme file:
$ npm install -g create-react-native-app
$ create-react-native-app my-app
2) Copy src, App.json and package.json from existing project into the new app.
3) In app.json, change the SDK version to 25.0.0 to avoid [this error](https://github.com/expo/expo/issues/1595).
3) Delete node_modules and then run npm install.
4) Run the simulator with npm start or react-native run-ios.

### Expected Behavior

I expected I would be able to re-use code from an existing app in this newly created app.

### Actual Behavior

The error, as mentioned above.  App will not run.  See screenshot:

![simulator screen shot - ipad air 2 - 2018-04-02 at 16 59 31](https://user-images.githubusercontent.com/1728052/38215923-39f02652-3697-11e8-99a0-0d676ddb85ab.png)

**Comment by arnoldbird** [comment]
Here is my package.json:

{
  "name": "Ally",
  "version": "0.1.0",
  "private": true,
  "devDependencies": {
    "babel-eslint": "^8.2.2",
    "eslint": "^4.18.2",
    "eslint-plugin-react": "^7.7.0",
    "flow-bin": "^0.66.0",
    "jest-expo": "25.0.0",
    "react-native-scripts": "1.11.1",
    "react-test-renderer": "16.2.0"
  },
  "main": "./node_modules/react-native-scripts/build/bin/crna-entry.js",
  "scripts": {
    "start": "react-native-scripts start",
    "eject": "react-native-scripts eject",
    "android": "react-native-scripts android",
    "ios": "react-native-scripts ios",
    "test": "node node_modules/jest/bin/jest.js"
  },
  "jest": {
    "preset": "jest-expo"
  },
  "dependencies": {
    "expo": "^25.0.0",
    "fetchival": "^0.3.3",
    "flow-typed": "^2.3.0",
    "install": "^0.10.4",
    "lodash": "^4.17.5",
    "node-fetch": "^2.0.0",
    "npm": "^5.6.0",
    "prop-types": "^15.6.0",
    "react": "16.2.0",
    "react-native": "https://github.com/expo/react-native/archive/sdk-25.0.0.tar.gz",
    "react-native-datepicker": "^1.6.0",
    "react-native-elements": "^0.19.0",
    "react-native-gesture-handler": "^1.0.0-alpha.40",
    "react-native-hyperlinked-text": "0.0.8",
    "react-native-keyboard-aware-scroll-view": "^0.4.3",
    "react-native-material-design": "^0.3.7",
    "react-native-responsive-dimensions": "^1.0.2",
    "react-native-vector-icons": "^4.5.0",
    "react-navigation": "git+https://github.com/react-community/react-navigation.git",
    "tcomb-form-native": "^0.6.11"
  }
}

**Comment by nikki93** [comment]
Seems it's from your 'react-navigation' import? Can you reproduce this in snack.expo.io with the minimal code / external dependencies possible?

**Comment by arnoldbird** [comment]
Some more context:

I had an app that worked fine.  I wanted to eject.  In the past, to eject, I commit latest changes, and then checkout a new branch ("eject").  I can then eject, and can still checkout master to use the app in Expo.  

In the past that worked, but not this time.  This time, when I checked out master, deleted node_modules and ran npm install, the install worked, but then running the project in simulator did not work.

So I decided to create a new app with create-react-native-app and copy in my custom code (App.json and src) and also package.json.  I figured I would then be able to delete node_modules, run npm install, and I would have my old expo app back.  But instead I am getting the errors I've reported today, related to the Expo SDK version and also the TabNavigator error.

**Comment by nikki93** [comment]
Yeah I think it would help to write import of TabNavigator from scratch in a new project to narrow down the issue. Right now your issue description contains everything except the actual code causing the issue, so it's hard to figure it out...

Also note that 'TabNavigator' is from 'react-navigation', not from 'react-native'.

**Comment by nikki93** [comment]
It "should be able to," but now it isn't, so this discussion is now about narrowing down what the error is. Which could be helped by looking for the import of 'TabNavigator' in your app and seeing what's going on there.

**Comment by nikki93** [comment]
FWIW the new project template imports TabNavigator correctly.

**Comment by arnoldbird** [comment]
Thanks again.  Here are the first 8 lines of my App.js, which contain the import.  This import has worked consistently for a few months, and is unchanged...

```
/**
 * @flow
 */

import React from 'react'
import { View } from 'react-native'

import { TabNavigator, StackNavigator } from 'react-navigation'
```

**Comment by arnoldbird** [comment]
@nikki93 

You're treating this like code needs to be debugged, but I think it's my workflow that needs to be debugged.  The code is unchanged.  I can't produce a snack representing the workflow, I don't think.  Would a snack provide insight into what's wrong with my workflow?  If so, I should make a snack.  

I just figure there must be an acceptable way to move code (src, App.js, package.json) from one expo app into another?

**Comment by arnoldbird** [comment]
I am encountering so many issues in the past 24 hours that it's hard to decide which, if any, should be closed in favor of others.  One issue is that Expo* thinks something is deprecated that is in fact not deprecated (I don't think).  For that issue, I have posted a question here: https://stackoverflow.com/questions/49631561/is-tabnavigator-deprecated

* Though I'm not sure if it is Expo or something else that is detecting the problem, or what appears to it to be a problem.

**Comment by quinlanj** [comment]
@arnoldbird Maybe something happened to your dependencies. TabNavigator shouldnt be deprecated, try `rm -rf node_modules` and building the dependencies again?

**Comment by arnoldbird** [comment]
@quinlanj Thanks for the response.  Yes, I've done that many times in the past 36 hours.  Though I don't need the -f flag.  But yes, I've deleted node_modules and rebuilt (npm install) a number of times.  Here is a screenshot with the terminal output telling me TabNavigator is deprecated.  Any idea what code is producing this output and what it might mean, since obviously TabNavigator is not deprecated?

![screen shot 2018-04-03 at 6 41 50 pm](https://user-images.githubusercontent.com/1728052/38279582-ecbb3ad8-376e-11e8-8856-1fc2b3db2b48.png)

**Comment by brentvatne** [comment]
you have installed react-navigation v2 somehow. if you install v1 you won't get this warning. v1 is the latest version on npm, see 1.5.9 here: https://www.npmjs.com/package/react-navigation

do you know how to do that or can i help you further with installing the package? let me know

**Comment by arnoldbird** [comment]
Brent,

That's it!  I had this in my package.json...

"react-navigation": "git+https://github.com/react-community/react-navigation.git",

I think I set it up this way a month or two ago because some issue I was dealing with required the latest code.  I didn't think much about it since then.  I guess the above annotation started grabbing v2 yesterday*.  I didn't even know there was a v2.

I switched to...

"react-navigation": "1.5.9",

...then deleted node_modules and rebuilt.  All good now.

*Another factor is that I rebuilt the app from scratch yesterday with creaate-react-native-app -- don't know if that would also tend to pull in v2 somehow.

Thanks.

**Comment by nickjuntilla** [comment]
Are createBottomTabNavigator and createStackNavigator deprecated from react-navigation? The both appear undefined in when importing from react-navigation 1.5.11

**Comment by nickjuntilla** [comment]
Solved my problem. I had to upgrade to "react-navigation": "^2.13.0",

**Comment by GeorgeFernandez** [comment]
Solved my Problem
I downgrade react navigation I changed the package.json as -- react-navigation": "^1.5.8
& I changed package-lock.json as well.

**Comment by kunalshah4345** [comment]
change version of react-navigation to 1.6.1 in package.json file
Delete node_modules & package-lock.json
Delete app from phone
run npm start -- --reset-cache
run app
